src/mds/Locker.cc

```diff
@@ -5138,7 +5140,8 @@ void Locker::scatter_writebehind(ScatterLock *lock)
 
   // forcefully take a wrlock
   lock->get_wrlock(true);
-  mut->emplace_lock(lock, MutationImpl::LockOp::WRLOCK);
+  auto it = mut->emplace_lock(lock, MutationImpl::LockOp::WRLOCK);
+  ceph_assert(it->is_wrlock());
```

Comment:
shouldn't this be

```suggestion
  it->flags |= MutationImpl::LockOp::WRLOCK;
```

as you do in `wrlock_force`?

Reply:
I didn't see the `scatter_writebehind` case will do a remote `wrlock`, but I added this check to make sure when it really doing this in future we can find the root cause asap.

Reply:
Ack. I just got suspicious because of the comment two lines above `// forcefully take a wrlock`. Then the assert is good, we'll quickly know if it happens